api/tokens.go

```diff
@@ -124,52 +123,35 @@ func (api *API) TokensHandler(ctx context.Context) http.HandlerFunc {
 }
 
 //SignOutHandler invalidates a users access token signing them out and returns a http handler interface
-func (api *API) SignOutHandler(ctx context.Context) http.HandlerFunc {
-	return func(w http.ResponseWriter, req *http.Request) {
-		var errorList []apierrorsdeprecated.Error
-
-		authString := req.Header.Get(AccessTokenHeaderName)
-		if authString == "" {
-			invalidTokenErrorBody := apierrorsdeprecated.IndividualErrorBuilder(apierrorsdeprecated.InvalidTokenError,
-				apierrorsdeprecated.MissingTokenDescription)
-			errorList = append(errorList, invalidTokenErrorBody)
-			log.Event(ctx, "no authorization header provided", log.ERROR)
-			errorResponseBody := apierrorsdeprecated.ErrorResponseBodyBuilder(errorList)
-			apierrorsdeprecated.WriteErrorResponse(ctx, w, http.StatusBadRequest, errorResponseBody)
-			return
-		}
+func (api *API) SignOutHandler(w http.ResponseWriter, req *http.Request, ctx context.Context, errorResponse *models.ErrorResponse) {
+	accessToken := models.AccessToken{
+		AuthHeader: req.Header.Get(AccessTokenHeaderName),
+	}
+	validationErr := accessToken.Validate(ctx)
+	if validationErr != nil {
```

Comment:
I think there would be some value in simplifying a step further, towards something like:

```suggestion
	if validationErr != nil {
	    return ErrorResponse{
	        Errors: validationErr,
	        Status: http.StatusBadRequest
	        }
	   }
```
- As soon as we encounter an error we should look to return, so theres no need to wait and do the len(errors) check in a separate step
- Since our handlers are all wrapped in baseHandler, we can actually return errors from here, rather than have to pass the additional errorResponse argument in - save the memory allocation until its actually needed.

Reply:
Changed to return a new ErrorResponse when an error is created.